File: practica/practica.java
public class practica {

    public static void main(String[] args) {
        int numero = 29; // Puedes cambiar este número para probar otros valores
        boolean esPrimo = esNumeroPrimo(numero);

        if (esPrimo) {
            System.out.println(numero + " es un número primo.");
        } else {
            System.out.println(numero + " no es un número primo.");
        }
    }

    public static boolean esNumeroPrimo(int numero) {
        // Los números menores que 2 no son primos
        if (numero < 2) {
            return false;
        }

        // Comprobar divisibilidad desde 2 hasta la raíz cuadrada del número
        for (int i = 2; i <= Math.sqrt(numero); i++) {
            if (numero % i == 0) {
                return false; // Si encontramos un divisor, el número no es primo
            }
        }

        return true; // Si no encontramos divisores, el número es primo
    }
}

File: practica/practica.rb
def practica_numeros_primos(numero)
  # Los números menores que 2 no son primos
  return false if numero < 2
  
  # Comprobar divisibilidad desde 2 hasta la raíz cuadrada del número
  (2..Math.sqrt(numero).to_i).each do |i|
    return false if numero % i == 0  # Si encontramos un divisor, el número no es primo
  end

  true  # Si no encontramos divisores, el número es primo
end

# Definir un número para comprobar
numero = 29  # Puedes cambiar este número para probar otros valores
es_primo = practica_numeros_primos(numero)

if es_primo
  puts "#{numero} es un número primo."
else
  puts "#{numero} no es un número primo."
end
